src/pathways/tags/tags.service.ts

```
@@ -387,14 +424,59 @@ export class TagsService implements OnModuleInit {
    * Note: Archived tags are excluded by default unless explicitly requested
    */
-  async list(listTagDto: ListTagDto, response: Response): Promise<Response> {
+  async list(listTagDto: ListTagDto, tenantId: string, response: Response): Promise<Response> {
     const apiId = APIID.TAG_LIST;
+    const tagListCacheTtl = parseInt(
+      this.configService.get('TAG_LIST_CACHE_TTL_SECONDS') || '1800',
+      10
+    );
     try {
+      const cacheKey = this.generateTagListCacheKey(listTagDto);
+      const cachedResult = await this.cacheService.get<{ count: number; limit: number; offset: number; items: Tag[] }>(cacheKey);
```

ESLint

[error] Replace `·count:·number;·limit:·number;·offset:·number;·items:·Tag[]` with `⏎········count:·number;⏎········limit:·number;⏎········offset:·number;⏎········items:·Tag[];⏎·····`

(prettier/prettier)